projects/cds/src/profiletag/services/profile-tag.injector.ts

```diff
@@ -93,6 +103,29 @@ export class ProfileTagInjector {
       );
   }
 
+  /**
+   * Listens to the changes to the cart and pushes the event for profiletag to pick it up further.
+   */
+  private cartChanged(): Observable<boolean> {
+    return combineLatest([
+      this.cartService.getEntries(),
+      this.cartService.getActive(),
+    ]).pipe(
+      skipWhile(([entries]) => entries.length === 0),
```

Comment:
Note: this will skip while the cart has not entries. As soon as there are entries in the cart, if will not skip. But, if the user clears the cart (so the cart now has zero entries), it will _not_ skip it _again_.
To test this, add a console log in the `tap`, and try adding an item to the cart, and then remove it.

Was this your intent?

Reply:
I guess yes, because you covered this case with a test: https://github.com/SAP/cloud-commerce-spartacus-storefront/pull/5533/files#diff-fd102ca47914438771221bd44c621247R298

Reply:
exactly. We noticed that the first event in a pipe is always an empty cart, and we wanted to avoid the situation where we send a full cart, someone does a page refresh, then we send an empty cart which is then followed by the previous full cart. In the case that someone is sitting on a page, and then goes from a full cart to an empty cart, this would be valid as it would signal that the items were removed. I.E. we need to distinguish between a removed from cart and a page re-fresh